File: src/kura/file/block_id.py
class BlockID:
    filename: str
    block_num: i64

    def __init__(self, filename: str, block_num: i64):
        self.filename = filename
        self.block_num = block_num

    def __eq__(self, other: object) -> bool:
        if not isinstance(other, BlockID):
            return NotImplemented
        return self.filename == other.filename and self.block_num == other.block_num

    def __str__(self) -> str:
        return f"{self.filename}.{self.block_num}"

    def __hash__(self) -> int:
        return hash(str(self))

File: internal/file/block_id.go
package file

import (
	"fmt"
	"hash/fnv"
)

type BlockID struct {
	filename string
	blockNum int64
}

func NewBlockID(filename string, blockNum int64) *BlockID {
	return &BlockID{
		filename: filename,
		blockNum: blockNum,
	}
}

func (b *BlockID) Filename() string {
	return b.filename
}

func (b *BlockID) Number() int64 {
	return b.blockNum
}

func (b *BlockID) String() string {
	return fmt.Sprintf("[file %s, block %d]", b.filename, b.blockNum)
}

func (b *BlockID) Hash() uint32 {
	h := fnv.New32a()
	_, _ = h.Write([]byte(b.String()))
	return h.Sum32()
}
